src/vm/checker/typecheck/interface.rs

```diff
@@ -0,0 +1,221 @@
+use std::collections::BTreeMap;
+
+use vm::types::{TypeSignature, FunctionArg, AtomTypeIdentifier, TupleTypeSignature};
+use vm::checker::typecheck::FunctionType;
+
+#[derive(Debug, Serialize, Clone)]
+pub enum ContractInterfaceFunctionAccess {
+    private,
+    public,
+    read_only,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ContractInterfaceTupleType {
```

Comment:
This is probably more accurately named `ContractInterfaceTupleEntryType`, since a tuple type is actually a Vec of these.

Reply:
Good catch. Fixed.